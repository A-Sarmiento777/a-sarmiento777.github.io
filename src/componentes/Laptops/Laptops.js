import React, { useEffect, useState } from 'react';
//import { apiURL } from '../../Utils/ApiUrl';
import Laptop from '../Laptop/Laptop';

const Laptops = ({agregarProductoAlCarrito, agregarProductoAlCarritoFav, removeItemFromCartFav, cart, favProducts}) => {
	const [laptops, setLaptops] = useState([])

	useEffect(() => {
		async function getLaptops() {
			try {
				//let response = await simpleFetch(apiURL + 'api/laptops')
				let response = await simpleFetch('/api/laptops')
				if (response) {
					setLaptops(response)
				}
			} catch (error) {
				console.log(error)
			}
		}
		getLaptops()
		// eslint-disable-next-line react-hooks/exhaustive-deps
	}, [])

	async function simpleFetch(url) {
		return await (await fetch(url)).json();
	}
	return (
		<div className="pt-4 text-center"> 
			<h1>Laptops</h1>
			
			<div className="d-flex flex-wrap justify-content-center">
			{laptops && laptops.map( (laptop, index) => {
				return (
					<Laptop agregarProductoAlCarrito={agregarProductoAlCarrito} agregarProductoAlCarritoFav={agregarProductoAlCarritoFav} removeItemFromCartFav={removeItemFromCartFav} cart={cart} favProducts={favProducts} laptop={laptop}/>
				)
			})}
			</div>
			</div>
	);
}
 
export default Laptops;