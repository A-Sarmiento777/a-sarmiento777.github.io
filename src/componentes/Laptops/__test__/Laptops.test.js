import React from "react";
import { act, create, waitFor } from "react-test-renderer";
import Laptops from "../Laptops";

test("It should render completed Laptops", async () => {
  jest.spyOn(global, "fetch").mockImplementation(() => {
    return Promise.resolve({
      json: () => Promise.resolve([]),
    });
  });

  let testRenderer;
  await act(async () => {
    testRenderer = create(
      <Laptops priceFormatter={() => {}} agregarProductoAlCarrito={() => {}} agregarProductoAlCarritoFav={() => {}} removeItemFromCartFav={() => {}} cart={[]} favProducts={[]} />
    );
  });

  let tree = testRenderer.toJSON();
  expect(tree).toMatchSnapshot();
  expect(global.fetch).toHaveBeenCalledTimes(1);
});
