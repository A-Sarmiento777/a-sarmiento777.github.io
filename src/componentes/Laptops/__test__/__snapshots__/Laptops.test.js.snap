// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`It should render completed Laptops 1`] = `
<div
  className="pt-4 text-center"
>
  <h1>
    BÃ¤rbara Datorer
  </h1>
  <div
    className="d-flex flex-wrap justify-content-center"
  />
</div>
`;
